### PR TITLE
debug: clean up fetching of the contents of /proc/device-tree

### PR DESCRIPTION
**Pull request by WavyEbuilder**
#### Describe your PR, what does it fix/add?

Works a bit more on https://github.com/hyprwm/Hyprland/issues/8400, should basically get done what can reasonably be cleaned up for HyprCtl.cpp

#### Is there anything you want to mention? (unchecked code, possible bugs, found problems, breaking compatibility, etc.)

I attempted to match the behavior of cat again, but I'm not sure if you want an error printing if `/proc/soc*/gpu*` can't be matched. Happy to add that if desired.

#### Is it ready for merging, or does it need work?

Should be ready.

### CODE REVIEW

File: src/debug/HyprCtl.cpp [hunk 1]
```diff
@@ -945,7 +947,22 @@ std::string systemInfoRequest(eHyprCtlOutputFormat format, std::string request)
 #if defined(__DragonFly__) || defined(__FreeBSD__)
     const std::string GPUINFO = execAndGet("pciconf -lv | fgrep -A4 vga");
 #elif defined(__arm__) || defined(__aarch64__)
-    const std::string GPUINFO = execAndGet("cat /proc/device-tree/soc*/gpu*/compatible");
+    std::string                 GPUINFO;
+    const std::filesystem::path dev_tree = "/proc/device-tree";
+    if (std::filesystem::exists(dev_tree) && std::filesystem::is_directory(dev_tree)) {
```
Comment (vaxerski) [inline src/debug/HyprCtl.cpp:952]: all of this might throw so probably should be in a try catch

Comment (WavyEbuilder) [reply src/debug/HyprCtl.cpp:952]: The entire `if` block? Just catch `...` (and I guess like `execAndGet` set `GPUINFO` to `"error"`) okay?

Comment (vaxerski) [reply src/debug/HyprCtl.cpp:952]: every std::filesystem call can throw unless supplied with a std::error_code as the second arg, so the entire thing

Comment (WavyEbuilder) [reply src/debug/HyprCtl.cpp:952]: Alright, will do

Comment (WavyEbuilder) [reply src/debug/HyprCtl.cpp]: Done

**Pull request merged**